packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -82,7 +82,9 @@ export class RouterliciousRestWrapper extends RestWrapper {
                     isNetworkError ? `NetworkError: ${error.message}` : safeStringify(error));
             }));
 
-        const responseBody: any = await response.clone().json().catch(async () => response.text());
+        const responseBody: any = response.headers.get("content-type")?.includes("application/json")
```

Comment:
Is this logic applicable both to success & failure paths?
I do not know, (and not sure if ODSP gets it right), but glanced at ODSP - we never check content-type:
- on success the caller describes what to expect (it's either JSON or binary)
- on failure, we always use text.

Reply:
Took a quick look at the R11s APIs. Content-Type header _should_ be accurate, including for error paths. However, on line 110 here it looks like we're stringifying JSON errors anyways, so errors end up being strings no matter what. Probably safest bet would be to put this logic within the Success `if()` block, then parse the body stream as `.text()` around line 109 instead of stringifying an object.

Reply:
It looks like it checks `responseBody.retryAfter` if it's a 429

Reply:
Side note - can we add a real type annotation rather than settling for `any`?  It can serve as documentation for what we expect (and depend on) over the wire all in one place.  I think it'd be `const responseBody: string | { retryAfter?: number }`.  Even safer would be to write a helper to extract the response body, and have it do additional runtime type checking before returning a thing of that type.  But I fully accept that these additiona runtime checks may be overkill and it works without it.  But I always love getting rid of `any`.

Reply:
I think it would be `T | { retryAfter?: number }`